src/test/java/uk/gov/homeoffice/digital/sas/balancecalculator/kafka/consumer/BalanceCalculatorConsumerServiceTest.java

```diff
@@ -0,0 +1,144 @@
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.time.LocalDateTime;
+import java.util.UUID;
+
+@SpringBootTest
+@ExtendWith({OutputCaptureExtension.class})
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
+class BalanceCalculatorConsumerServiceTest {
+
+  @Autowired
+  BalanceCalculatorConsumerService balanceCalculatorConsumerService;
+
+  @Test
+  void onMessage_deserializeKafkaMessageAndLogSuccess_when_validMessageIsReceived
+      (CapturedOutput capturedOutput) throws JsonProcessingException, ClassNotFoundException {
+
+    // given
+    String id = UUID.randomUUID().toString();
+    String ownerId = UUID.randomUUID().toString();
+
+    TimeEntry expectedTimeEntry = TestUtils.createTimeEntry(
+        id,
+        ownerId,
+        TestUtils.getAsDate(LocalDateTime.now()),
+        TestUtils.getAsDate(LocalDateTime.now().plusHours(1)));
+
+    String message = TestUtils.createKafkaMessage(MESSAGE_VALID_RESOURCE, MESSAGE_VALID_VERSION,
+        id, ownerId);
+
+    //when
+    balanceCalculatorConsumerService.onMessage(message);
+
+    // then
+    assertThat(capturedOutput.getOut()).contains(String.format(KAFKA_SUCCESSFUL_DESERIALIZATION,
+        expectedTimeEntry.getId()));
+  }
+
+  //Invalid resource throws error
+  @Test
+  void onMessage_notDeserializeKafkaMessageAndThrowException_when_inValidResourceIsReceived() throws JsonProcessingException {
+    //given
+    String id = UUID.randomUUID().toString();
+    String ownerId = UUID.randomUUID().toString();
+
+    String message = TestUtils.createKafkaMessage(MESSAGE_INVALID_RESOURCE, MESSAGE_VALID_VERSION
+        , id, ownerId);
+
+
+    assertThatThrownBy(() -> {
+     balanceCalculatorConsumerService.onMessage(message);
+    }).isInstanceOf(KafkaConsumerException.class)
+        .hasMessageContaining(String.format(KAFKA_UNSUCCESSFUL_DESERIALIZATION, message));
+
+  }
+
+  //Invalid version throws error
+  @Test
+  void onMessage_notDeserializeKafkaMessageAndThrowException_inValidVersionIsReceived() throws JsonProcessingException {
+    String id = UUID.randomUUID().toString();
+    String ownerId = UUID.randomUUID().toString();
+
+    String message = TestUtils.createKafkaMessage(MESSAGE_VALID_RESOURCE, MESSAGE_INVALID_VERSION
+        , id, ownerId);
+
+    assertThatThrownBy(() -> {
+      balanceCalculatorConsumerService.onMessage(message);
+    }).isInstanceOf(KafkaConsumerException.class)
+        .hasMessageContaining(String.format(KAFKA_SCHEMA_INVALID_VERSION, message));
+  }
+
+  //Desearilization error (missing/extra field?) on resource
+  @Test
+  void onMessage_notDeserializeKafkaMessage_extraFieldReceived() throws JsonProcessingException {
+    String id = UUID.randomUUID().toString();
+    String ownerId = UUID.randomUUID().toString();
+
+    ObjectMapper mapper = new ObjectMapper();
```

Comment:
make this mapper a class member so it doesn't have to be instantiated for different tests